pkg/logs/client/http/destination.go

```diff
@@ -376,15 +379,58 @@ func (d *Destination) updateRetryState(err error, isRetrying chan bool) bool {
 }
 
 func httpClientFactory(timeout time.Duration, cfg pkgconfigmodel.Reader) func() *http.Client {
+
+	transport := httputils.CreateHTTPTransport(cfg)
+
+	// Configure transport based on user setting
+	switch cfg.Get("logs_config.transport_type") {
+	case "http1":
+		transport.TLSNextProto = make(map[string]func(authority string, c *tls.Conn) http.RoundTripper)
+		transport.TLSClientConfig = &tls.Config{
+			NextProtos: []string{"http/1.1"},
+		}
+	case "auto":
+		// Use default ALPN auto-negotiation
+	default:
+		log.Warnf("Invalid transport_type '%s', falling back to 'auto'", cfg.GetString("logs_config.transport_type"))
+		// Use default ALPN auto-negotiation
+	}
+
 	return func() *http.Client {
-		return &http.Client{
+		client := &http.Client{
 			Timeout: timeout,
 			// reusing core agent HTTP transport to benefit from proxy settings.
-			Transport: httputils.CreateHTTPTransport(cfg),
+			Transport: transport,
 		}
+
+		log.Tracef("Log Agent is using %v transport for connection", getTransportProtocol(client))
+		return client
 	}
 }
 
+// getTransportProtocol return the transport type
+func getTransportProtocol(client *http.Client) string {
```

Comment:
I don't think we know for sure what protocol is being used until we make the first request and establish a connection. 

I'm also not sure we need the log line this is used in since we will log the protocol with each payload that is posted. So IMO you can remove this.